app/src/main/jni/algorithm_module.hpp

```diff
@@ -45,6 +45,9 @@ class AlgorithmModule {
     virtual std::string status() const { return ""; }
     virtual int trackingStatus() { return -1; };
 
+    virtual void writeInfoFile(const std::string &mode, const std::string &device) {};
```

Comment:
This is getting messy. I would prefer adding this kind of stuff to this API, which tells the static device information to the recorder or algorithm module:
```c++
struct DeviceInfo {
    std::string name;
    typedef std::array<std::array<double, 3>, 3> RotationMatrix;
    RotationMatrix imuToCameraMatrix; // always the same for Android, could also be omitted
    float focalLength = -1;
    // ...
};

void setDeviceInfo(const DeviceInfo &info);
```
We can then either modify the JSONL recorder to allow stuff like
```c++
jsonlRecorder.writeExtraJsonFile("info.json", jsonObject);
// (and maybe also this)
jsonlRecorder.writeExtraTextFile("parameters.txt", someString);
```
or just dump the device info and/or extra parameters into the main JSONL file
```
jsonlRecorder.writeJson({ info: { ... }});
```
OR
```
jsonlRecorder.writeDeviceInfo(deviceInfo);
```
in which case the DeviceInfo would need to be added to the jsonl-recorder types, which seems a bit clunky